common/src/main/java/io/netty5/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -15,33 +15,28 @@
  */
 package io.netty5.util.concurrent;
 
-import io.netty5.util.internal.DefaultPriorityQueue;
-import io.netty5.util.internal.PriorityQueue;
 import io.netty5.util.internal.PriorityQueueNode;
 
-import java.util.Comparator;
-import java.util.Queue;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.Executors.callable;
 
 /**
  * Abstract base class for {@link EventExecutor}s that want to support scheduling.
  */
 public abstract class AbstractScheduledEventExecutor extends AbstractEventExecutor {
 
-    private static final Comparator<RunnableScheduledFutureNode<?>> SCHEDULED_FUTURE_TASK_COMPARATOR =
-            Comparable::compareTo;
-    private static final RunnableScheduledFutureNode<?>[]
-            EMPTY_RUNNABLE_SCHEDULED_FUTURE_NODES = new RunnableScheduledFutureNode<?>[0];
-
-    private PriorityQueue<RunnableScheduledFutureNode<?>> scheduledTaskQueue;
+    private TaskScheduler taskScheduler;
+    private TaskSchedulerFactory taskSchedulerFactory;
```

Comment:
**Suggestion**: Consider documenting the behavior when both taskScheduler and taskSchedulerFactory are null. It might be beneficial to clarify the expected behavior or fallback mechanism in such cases, ensuring that users of the API have a clear understanding of how task scheduling is handled. <br> **Code Suggestion**: 
 ```
 +    /**
 +     * Task scheduler instance used for scheduling tasks. If null, the taskSchedulerFactory is used
 +     * to create a new instance. If both are null, document the expected behavior or fallback mechanism.
 +     */
 +    private TaskScheduler taskScheduler;
 +    private TaskSchedulerFactory taskSchedulerFactory;
 ```